Development/sdp/sdp_grammar.cpp

```diff
@@ -714,6 +714,54 @@ namespace sdp
                 {
                     sdp::attributes::mediaclk,
                     string_converter // sorry, cannot summon the energy
+                },
+                {
+                    sdp::attributes::extmap,
+                    {
+                        [](const web::json::value& v) {
+                            std::string s;
+                            s += number_converter.format(v.at(sdp::fields::local_id));
+                            if (v.has_field(sdp::fields::direction)) s += "/" + string_converter.format(v.at(sdp::fields::direction));
+                            s += " " + string_converter.format(v.at(sdp::fields::uri));
+                            if (v.has_field(sdp::fields::extensionattributes)) s += " " + string_converter.format(v.at(sdp::fields::extensionattributes));
+                            return s;
+                        },
+                        [](const std::string& s) {
+                            auto v = web::json::value::object(keep_order);
+                            size_t pos = (!s.empty() && ' ' == s.front()) ? 1 : 0;
```

Comment:
The ABNF doesn't seem to allow SP before the local ID? Are bad implementations common in the wild?
https://www.rfc-editor.org/rfc/rfc5285#section-7

I wonder if the rest of this parsing function can be done more simply, I'll have a play and see!

Reply:
Yes, SP is not allowed and I never seen such SDP samples, it's my bad.

Regarding making it simpler, I see that the regex may be turned into just `\D`. As for the optional `direction` syntax element, the only alternative that I see now is reading a string token and searching it in the list of directions but I'm not sure that it's simpler than the current implementation.